### PR TITLE
Document OS and software support strategy for virtual-environments

### PR DESCRIPTION
**Pull request by AlenaSviridenko**
# Description
We still don't have any documented description of images and software support and preinstallation policy. It was not clear for community how we decide what to preinstall, how many versions, when we change defaults and so on. 
I collected a set of guidelines that we are already following or will follow for the upcoming requests.

🎨[Rendered version](https://github.com/AlenaSviridenko/virtual-environments/blob/document-software-support-policy/docs/software-and-images-guidelines.md) 🎨

Also, renamed folder `helpers` to `docs` and changed documents name fomatting

## Check list
- [n\a] Related issue / work item is attached
- [n\a] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [n\a] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by miketimofeev — APPROVED** [review]

File: docs/software-and-images-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+In general, these are the guidelines we consider when deciding what to pre-install:
+
+- Tools and ecosystems that are broadly popular and widely-used will be given priority.
+- Recent versions of tools will be given priority over older versions.
+- Tools and versions that are deprecated or have reached end-of-life will not be added.
+- If a tool can be installed during the build, we will evaluate how much time is saved
+ and how much space is used by having the tool pre-installed.
+- MIT, Apache, and GNU licenses are ok, anything else we'll have to check with lawyers.
+- If a tool takes much space we will evaluate space usage and provide a decision if this tool can be pre-installed.
+- If a tool requires the support of more than one version, we will consider the cost of this maintenance, how often new versions bring dangerous updates.
```
Comment (maxim-lobanov) [inline docs/software-and-images-guidelines.md:13]: 2 general nitpicks:
1) Some lines have `.` at the end. Some don't have. Let's do them consistent.
2) May be rewrite sentences that started with `If` to be consistent with other ones.

File: docs/software-and-images-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+| Tool name | Installation strategy |
+|-----------|-----------------------|
+| Java      | all LTS versions |
+| Node.js   | 3 latest LTS versions |
+| Go        | 3 latest minor versions     |
+| Python <br/> Ruby | 5 most popular `major.minor` versions |
+| PyPy      | 3 most popular `major.minor` versions |
+| .NET Core | 2 latest LTS versions and 1 latest version. For each feature version only latest patch is installed |
+| GCC <br/> GNU Fortran <br/> Clang <br/> GNU C++ | 3 latest major versions |
+| Android NDK | 1 latest, 1 LTS version |
```
Comment (maxim-lobanov) [inline docs/software-and-images-guidelines.md:33]: I think we should also add `Xcode`

File: docs/software-and-images-guidelines.md (hunk 1 above)
Comment (AlenaSviridenko) [reply docs/software-and-images-guidelines.md:13]: I think, that some statements will look weird with the "If" in the beginning, so I'd leave this section as is.

**Review by johnperez416 — APPROVED** [review]

**Review by johnperez416 — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

File: docs/software-and-images-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- MIT, Apache, and GNU licenses are ok, anything else we'll have to check with lawyers.
+- If a tool takes much space we will evaluate space usage and provide a decision if this tool can be pre-installed.
+- If a tool requires the support of more than one version, we will consider the cost of this maintenance, how often new versions bring dangerous updates.
+
+**Note:** For new tools, please, create an issue and get an approval from us to add this tool to the image before creating the pull request.
+
+## Software and images support policy
+These are the guidelines we follow in software and images supporting routine:
+- Tools and versions will typically be removed 6 months after they are deprecated or have reached end-of-life.
+- We support at least 2 latest OS versions (LTS only for Ubuntu) and initiate deprecation process for the oldest one when image usage drops below 5%.
```
Comment (hutson) [inline docs/software-and-images-guidelines.md:20]: I am late to this PR, but... what does `oldest one` mean here? Old actively supported LTS? Or oldest version of Ubuntu currently offered by GitHub? When it states `initiate deprecation process ... drops below 5%` does that mean you could end up with `ubuntu-16.04` for decades if 6% of builds continue to use that label?

**Pull request merged**